RM-973 Release Notes 2.25.0 (#643)
percona/pmm-doc: docs/release-notes/2.25.0.md

# Percona Monitoring and Management 2.25.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">December 7, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## Release Highlights 
- **Percona Platform (Technology Preview):** 
    - Ability to connect your Percona Monitoring and Management instances to your Percona Platform to get the latest intelligence from Percona and have all your accounts and relations with Percona centralized. READ_MORE

- **Enhanced PostgreSQL monitoring**

>     - From this release, PMM will let users who add PostgreSQL Servers be monitored by PMM to [specify database name for monitoring](https://deploy-preview-643--pmm-doc.netlify.app/details/commands/pmm-admin.html#postgresql). Previous versions have the default `postgres` database, which wasn't always convenient for the users. 

Reviewer: ```suggestion
    - PMM will now let users who add PostgreSQL Servers be monitored by PMM to [specify database name for monitoring](https://deploy-preview-643--pmm-doc.netlify.app/details/commands/pmm-admin.html#postgresql). Previous versions have the default `postgres` database, which wasn't always desired. 

```

>     - This Release of PMM is supported and compatible with the recent release of the PostgreSQL extension - `pg_stat_monitor` v1.0.0. You can read more about its features and advantages in this blog post (LINK to pg_stat_monitor release announcement or Userguide)

Reviewer: ```suggestion
    -  Now that `pg_stat_monitor` has released it's GA v1.0.0, it is fully supported in PMM from this release forward with many new PostgreSQL metrics, Dashboards and Query Analytics features available!  You can read more about its features and advantages in this blog post (LINK to pg_stat_monitor release announcement or Userguide)

```

>     - PMM 2.25 also be compatible and working fine with the latest Percona Disctibutions for PostgreSQL released today (LINK), which includes the newest version of pg_stat_monitor extension.  

Reviewer: ```suggestion
    - PMM 2.25.0 is also now compatible with the latest Percona Distribution for PostgreSQL released today (LINK), which includes the newest version of pg_stat_monitor extension.  

```

Reviewer: just be aware that "released today" may not be accurate when we actually release ;-) might be better to just call out the specific distribution version so that in 2 months someone reading this doesn't jump to the wrong conclusion

- **Grafana usability enhancements**

>     - We changed the way how the dashboards, plugins, and data sources are added into PMM. We now use the native Grafana provisioning mechanism in PMM. This change will lead to a faster and more reliable update process for our users from one version of PMM to another. With this update, we'll preserve all custom changes you might already do with plugins and dashboards, but we'll encourage you to be back up them before an upgrade and recheck after, just in case. 

Reviewer: ```suggestion
    - We changed the way that dashboards, plugins, and data sources are added into PMM. We are now using the native Grafana provisioning mechanism in PMM. This change will lead to a faster and more reliable update process for our users moving from one version of PMM to another. With this update, we'll preserve all custom changes you have already done with plugins and dashboards, but we still encourage you to back them up before an upgrade and recheck after, just in case. 

```

    - This version of PMM also will let you change the time zone on dashboards. It will also preserve this selection while you navigate over Dashboards. If you need to change this setting permanently for your account - you need to change it in your preferences by the URL: https://YOUR_PMM_SERVER/graph/profile   

- **DBaaS (Technical Preview)**

>     - PMM will now let you deploy and update your PXC clusters created with DBaaS functionality to the version of Percona Distribution for MySQL Operator 1.9.0 to deliver the latest fixes in this version.

Reviewer: ```suggestion
    - You can now deploy and update your DBaaS created PXC clusters to the latest version of Percona Distribution for MySQL Operator 1.9.0 to take advantage of the latest features and fixes. 
```

Reviewer 2: it's not the latest ((  1.10 already released

- **PMM environment enhancements**

>     - For users who use PMM in automation, we added a pmm-client docker container to be started as a sidecar container and not fail if the connection to DB  is not available. Please see full details in 

Reviewer: Not sure I get why the "used in automation" is important here...

Reviewer 2: 🤔 .. in the usual case, you will start it manually and need to see - if the result was successful. No?

https://www.percona.com/doc/percona-monitoring-and-management/2.x/setting-up/client/index.html#install

>     - With the support of new products and new versions of already supported products, we also removed old software. Since this release, we are removing support of Ubuntu 16.04  in PMM. Please see more details in  

Reviewer: ```suggestion
    - With the support of new products and new versions of already supported products, we also removed old unsupported software. As of this release, we are no longer supporting Ubuntu 16.04 in PMM. Please see more details in  

```

- **Grafana usability enhancements**
    - We changed the way that dashboards, plugins, and data sources are added into PMM. We are now using the native Grafana provisioning mechanism in PMM. This change will lead to a faster and more reliable update process for our users moving from one version of PMM to another. With this update, we'll preserve all custom changes you have already done with plugins and dashboards, but we still encourage you to back them up before an upgrade and recheck after, just in case. 

    - This version of PMM also will let you change the time zone on dashboards. It will also preserve this selection while you navigate over Dashboards. If you need to change this setting permanently for your account - you need to change it in your preferences by the URL: `https://YOUR_PMM_SERVER/graph/profile`

- **DBaaS (Technical Preview)**
    - You can now deploy and update your DBaaS created PXC clusters to the latest version of Percona Distribution for MySQL Operator 1.9.0 to take advantage of the latest features and fixes. 

- **PMM environment enhancements**

>     - For users who use PMM client in kubernetes or build automation around it, we added a pmm-client docker container to be started as a sidecar container and not fail if the connection to DB  is not available. Please see full details in [PMM Client documentation](https://www.percona.com/doc/percona-monitoring-and-management/2.x/setting-up/client/index.html#install)

Reviewer 3: ```suggestion
    - Added a pmm-client docker container relevant if you're using PMM client in Kubernetes or if you build automation around it.
This can be started as a sidecar container and will not fail if the connection to DB  is not available. For more details, see the [PMM Client documentation](https://www.percona.com/doc/percona-monitoring-and-management/2.x/setting-up/client/index.html#install)

```

Reviewer 2: the pmm-client docker was already in PMM. What we added - is an additional flag, so people can start it as sidecar

Reviewer: ```suggestion
     - The pmm-client docker container can now be started as a sidecar.  For users that use PMM client in Kubernetes or build automation around it, you can now start the client as a sidecar container.  The client will also gracefully handle any instances where the connection to DB  is not available. For more details, see the [PMM Client documentation](https://www.percona.com/doc/percona-monitoring-and-management/2.x/setting-up/client/index.html#install)
```

>     - With the support of new products and new versions of already supported products, we also removed old, unsupported software. As of this release, we are no longer supporting Ubuntu 16.04 in PMM [according to recent anouncements](https://www.percona.com/blog/eol-pmmv1-mongo40-ubuntu1604)

Reviewer 3: ```suggestion
    - Removed support for Ubuntu 16.04. With the support of new products and new versions of already supported products, we also removed old, unsupported software. As of this release, we are no longer supporting Ubuntu 16.04 in PMM [according to recent anouncements](https://www.percona.com/blog/eol-pmmv1-mongo40-ubuntu1604)

```

# Percona Monitoring and Management 2.25.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">December 7, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

Highlights. 

> ** (Technology Preview ) Ability to connect your Percona Monitoring and Management instances to your Percona Portal to get the latest intelligence from Percona and have all your accounts and relations with Percona centralized. READ_MORE

Reviewer 4: Better to use Percona Platform, or Percona Platform Portal instead of Percona Portal

> ** From this release, PMM will let users who add PostgreSQL Servers be monitored by PMM to specify database name for monitoring. Previous versions have the default `postgres` database, which wasn't always convenient for the users. 

Reviewer 2: ```suggestion
** From this release, PMM will let users who add PostgreSQL Servers be monitored by PMM to specify database name for monitoring. Previous versions have the default `postgres` database, which wasn't always convenient for the users. [Link to documantation](TBD)

```

** This version of PMM also will let you change the time zone on dashboards. It will also preserve this selection while you navigate over Dashboards. If you need to change this setting permanently for your account - you need to change it in your preferences by the URL: https://YOUR_PMM_SERVER/graph/profile   

** (Technical Preview) DBaaS: PMM will now let you deploy and update your PXC clusters created with DBaaS functionality to the version of Percona Distribution for MySQL Operator 1.9.0 to deliver the latest fixes in this version.

** For users who use PMM in automation, we added a pmm-client docker container to be started as a sidecar container and not fail if the connection to DB  is not available. Please see full details in 
https://www.percona.com/doc/percona-monitoring-and-management/2.x/setting-up/client/index.html#install

** With the support of new products and new versions of already supported products, we also removed old software. Since this release, we are removing support of Ubuntu 16.04  in PMM. Please see more details in  
https://www.percona.com/blog/eol-pmmv1-mongo40-ubuntu1604

 

## New Features

> - [PMM-9050](https://jira.percona.com/browse/PMM-9050): Connect PMM Server to Percona Portal for additional account info in PMM and value added content

Reviewer 4: Better to use Percona Platform, or Percona Platform Portal, instead of Percona Portal.

Author: @Hlmn-Rmr go ahead and correct ticket titles. I am going to run jira-bot and merge ticket changes couple of times in a next couple of days.

# Percona Monitoring and Management 2.25.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">December 7, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

Highlights. 
** (Technology Preview ) Ability to connect your Percona Monitoring and Management instances to your Percona Portal to get the latest intelligence from Percona and have all your accounts and relations with Percona centralized. READ_MORE

> ** From this release, PMM will let users who add PostgreSQL Servers be monitored by PMM to specify database name for monitoring. Previous versions have the default `postgres` database, which wasn't always convenient for the users. [Link to documantation](TBD)

Reviewer 2: ```suggestion
** From this release, PMM will let users who add PostgreSQL Servers be monitored by PMM to [specify database name for monitoring](https://deploy-preview-643--pmm-doc.netlify.app/details/commands/pmm-admin.html#postgresql). Previous versions have the default `postgres` database, which wasn't always convenient for the users. 

```

- [PMM-6937](https://jira.percona.com/browse/PMM-6937): Can't add PostgreSQL instances to PMM without `postgres` DB in PostgreSQL server (Thanks to Daniel Kowalewski for reporting this issue)
- [PMM-7447](https://jira.percona.com/browse/PMM-7447): Can't add into PMM instances of PostgreSQL with SCRAM-SHA-256 authentication
- [PMM-9085](https://jira.percona.com/browse/PMM-9085): PMM Server crashes after upgrading to 2.22 every 4 hours
- [PMM-9156](https://jira.percona.com/browse/PMM-9156): `pmm-agent` paths-base option not working for pmm2-client binary installation in PMM 2.23.0
- [PMM-8461](https://jira.percona.com/browse/PMM-8461): DBaaS: Confusing error when accessing DBaaS pages when it's disabled
- [PMM-8110](https://jira.percona.com/browse/PMM-8110): DBaaS: Registering K8s cluster with operators already installed can cause error
- [PMM-8694](https://jira.percona.com/browse/PMM-8694): Query Analytics: URLs in Query Analytics with a selected query and a timestamp range does not select the query
- [PMM-9227](https://jira.percona.com/browse/PMM-9227): Pagination Reset on QAN after Time Range change doesn't work, results in wrong results

> - [PMM-9298](https://jira.percona.com/browse/PMM-9298): PMM AMI image in 2.24.0 has only 8GB space for data and Volume Size Check fails while upgrading to 2.25.0

Reviewer 3: ## Known issue
After connecting PMM to Percona Platform, PMM occasionally shows a false permission issue notification, incorrectly suggesting that the connection could not be established due to missing permissions. Reload the page to remove the incorrect notification and confirm the connection.
